### DISCUSSION TITLE
Slicing of nested objects, outer object translucent, inner objects opaque

### DISCUSSION THREAD

**Discussion by LindyBalboa** [pyvista/pyvista#3661] [Q&A]
Greetings,

I have been playing around with Pyvista the last few days and have been quite impressed with how fast it can make some complex graphics, however I am struggling to achieve what I want. I am trying to create a simplified representation of tomographic data. What I would like is a gray translucent cube, within which there are several other opaque objects of different colors. Then I would like to show slices from different positions/orientations. I have tried using the scalar/opacity arrays of single objects but ran into interpolation issues, I tried using a combined multiblock but couldn't get the opacity to work. I have spent enough time messing about without a usable solution that I am hoping one of the experts might be able to create a working example relatively fast.

Thanks,
Lindy

**Comment by banesullivan** [comment]
I'm not totally sure what you're after but perhaps this is a start:

```py
import pyvista as pv

cube = pv.Box((-1.5, 1.5, -1.5, 1.5, -1.5, 1.5))

sphere = pv.Sphere(center=(-0.5, -0.5, 0)).delaunay_3d()
cone = pv.Cone(center=(0.5, 0.5, 0)).delaunay_3d()

p = pv.Plotter(notebook=0)
p.add_mesh(cube, color='grey', opacity=0.5)
p.add_mesh(sphere, color='red')
p.add_mesh(cone, color='blue')
p.enable_depth_peeling()  # Improves opacity
p.show()
```
<img width="624" alt="Screen Shot 2022-12-02 at 9 34 18 AM" src="https://user-images.githubusercontent.com/22067021/205341090-7218be35-2777-44ee-afd5-e227282f36d9.png">


```py
# Make and show slices
ss = sphere.slice_orthogonal()
cs = cone.slice_orthogonal()

p = pv.Plotter(notebook=0)
p.add_mesh(cube, color='grey', opacity=0.5)
p.add_mesh(ss, color='red')
p.add_mesh(cs, color='blue')
p.enable_depth_peeling()  # Improves opacity
p.show()
```
<img width="624" alt="Screen Shot 2022-12-02 at 9 36 13 AM" src="https://user-images.githubusercontent.com/22067021/205341072-0d8bb883-0902-4bf4-9ce8-02464d48bc24.png">

**Reply by LindyBalboa** [reply]
Thanks for taking the time to respond. Seems I didn't quite get my message across. I meant single planar cross sections that cuts through and show s a 2D image of all the actors in the scene, which could then be set at an arbitrary location/angle. So in the case you posted a red circle and blue triangle on a gray background.

I also tried with modifying the scalars of one large uniform mesh using select_enclosed_scalars and using volume rendering, but was struggling to get the opacity to work.

**Reply by banesullivan** [reply]
I'm still not totally sure what you're after but here is an interactive demo that may get you where you're hoping:

```py
import pyvista as pv
import vtk

cube = pv.Box((-1.5, 1.5, -1.5, 1.5, -1.5, 1.5))

sphere = pv.Sphere(center=(-0.5, 0.5, 0))
cone = pv.Cone(center=(0.5, 0.5, 0))

data_to_slice = pv.MultiBlock([cube, sphere, cone])

p = pv.Plotter(notebook=0)
p.add_mesh(cube, color='grey', opacity=0.5)
p.add_mesh(sphere, color='red')
p.add_mesh(cone, color='blue')

def slicer(normal, origin):
    # slice
    slc = data_to_slice.slice(normal=normal, origin=origin).combine()
    # move slice to Y plane
    slc.points[:,1] = -2.0
    p.add_mesh(slc, name='full-slice', color='k')
    

p.add_plane_widget(slicer, assign_to_axis='y', )

p.show()
```
<img width="624" alt="Screen Shot 2022-12-19 at 1 22 42 PM" src="https://user-images.githubusercontent.com/22067021/208514396-8424f229-22f4-4cb0-ac9b-5a0aa2f4af12.png">


You can combine geometries to slice together via the `MultiBlock` class, then `combine()` the `slice()` result and move/translate however you like
